runtime/src/accounts.rs

```diff
@@ -619,8 +619,13 @@ impl Accounts {
             .collect()
     }
 
-    pub fn calculate_capitalization(&self, ancestors: &Ancestors) -> u64 {
-        self.accounts_db.unchecked_scan_accounts(
+    pub fn calculate_capitalization(&self, ancestors: &Ancestors, slot: Slot) -> u64 {
```

Comment:
should we also pass the `use_write_cache/can_cached_slot_be_unflushed` boolean here so callers have to be cognizant of whether they have left slots unflushed in the cache?